core/src/syn/parser/stmt/define.rs

```diff
@@ -1212,6 +1216,122 @@ impl Parser<'_> {
 		Ok(res)
 	}
 
+	pub async fn parse_define_config(
+		&mut self,
+		ctx: &mut Stk,
+	) -> ParseResult<DefineConfigStatement> {
+		let (if_not_exists, overwrite) = if self.eat(t!("IF")) {
+			expected!(self, t!("NOT"));
+			expected!(self, t!("EXISTS"));
+			(true, false)
+		} else if self.eat(t!("OVERWRITE")) {
+			(false, true)
+		} else {
+			(false, false)
+		};
```

Comment:
Since there are only three possible options here `IF NOT EXISTS` `OVERWRITE` or neither this state should probably be captured in an enum with three variants. Having two boolean would allow both to be true which is not allowed by this code.

Reply:
I'll do this in a seperate pr